src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp

```diff
@@ -90,51 +92,91 @@ class instance_trial_of_the_champion : public InstanceMapScript
         void OnCreatureCreate(Creature* creature) override
         {
             Map::PlayerList const &players = instance->GetPlayers();
-            uint32 TeamInInstance = 0;
-
             if (!players.isEmpty())
             {
                 if (Player* player = players.begin()->GetSource())
-                    TeamInInstance = player->GetTeam();
+                {
+                    if (GetData(DATA_PLAYERS_TEAM) == 0)
+                        SetData(DATA_PLAYERS_TEAM, player->GetTeam());
+                }
+                
             }
 
             switch (creature->GetEntry())
             {
-                // Champions
-                case VEHICLE_MOKRA_SKILLCRUSHER_MOUNT:
-                    if (TeamInInstance == HORDE)
-                        creature->UpdateEntry(VEHICLE_MARSHAL_JACOB_ALERIUS_MOUNT);
-                    break;
-                case VEHICLE_ERESSEA_DAWNSINGER_MOUNT:
-                    if (TeamInInstance == HORDE)
-                        creature->UpdateEntry(VEHICLE_AMBROSE_BOLTSPARK_MOUNT);
-                    break;
-                case VEHICLE_RUNOK_WILDMANE_MOUNT:
-                    if (TeamInInstance == HORDE)
-                        creature->UpdateEntry(VEHICLE_COLOSOS_MOUNT);
-                    break;
-                case VEHICLE_ZUL_TORE_MOUNT:
-                    if (TeamInInstance == HORDE)
-                        creature->UpdateEntry(VEHICLE_EVENSONG_MOUNT);
-                    break;
-                case VEHICLE_DEATHSTALKER_VESCERI_MOUNT:
-                    if (TeamInInstance == HORDE)
-                        creature->UpdateEntry(VEHICLE_LANA_STOUTHAMMER_MOUNT);
+                case NPC_MOKRA:
+                case NPC_ERESSEA:
+                case NPC_RUNOK:
+                case NPC_ZULTORE:
+                case NPC_VISCERI:
+                case NPC_JACOB:
+                case NPC_AMBROSE:
+                case NPC_COLOSOS:
+                case NPC_JAELYNE:
+                case NPC_LANA:
+                    SetGrandChampionData(creature);
                     break;
                 // Coliseum Announcer || Just NPC_JAEREN must be spawned.
                 case NPC_JAEREN:
                     uiAnnouncerGUID = creature->GetGUID();
-                    if (TeamInInstance == ALLIANCE)
+                    if (GetData(DATA_PLAYERS_TEAM) == ALLIANCE)
                         creature->UpdateEntry(NPC_ARELAS);
                     break;
-                case VEHICLE_ARGENT_WARHORSE:
-                case VEHICLE_ARGENT_BATTLEWORG:
+                case VEHICLE_ARGENT_WARHORSE_COSMETIC:
```

Comment:
"The cosmetic" vehicles are what you see unfriendly vehicles. On alliance they are wolves and on horde they are horses. These are used by Grand Champions if they get fallen out of mount. After falling out they try to reach these mounts. Normally in the instance there are spawned only alliance vehicles (unfriendly wolf and friendly horse). So if you enter in horde, we change their templates to horde vehicles (unfriendly horse and friendly wolf). Therefore we don't have to loop through all vehicles, only alliance vehicles. In addition, no matter are you horde or alliance the vehicles are added into vehiclelist.